server-side/controllers/user.controller.js

```diff
@@ -87,3 +50,262 @@ export const updatedUser = async (req, res,next) => {
+// function sendTemporaryPasswordToCustomerEmail(user) {
+//   const transporter = nodemailer.createTransport({
+//       service: 'gmail',
+//       auth: {
+//           user: 'mazaltovchagit16@gmail.com',
+//           pass: 'bysr woai nxmd jtti'
+//       }
+//   });
+
+//   const mailOptions = {
+//       from: 'mazaltovchagit16@gmail.com',
+//       to: "st3196420@gmail.com",
+//       subject: 'סיסמא זמנית ND',
+//       html: `
+//       <div dir="rtl">
+//       <p>הסיסמה הזמנית שלך היא: ${user.name}</p>
+//       </div>
+//   `
+//   };
+
+//   transporter.sendMail(mailOptions, function (error, info) {
+//       if (error) {
+//           console.error(error);
+//       }
+//   });
+
+// }
+
+// export const sendEmail = async (req, res) => {
+  
+//     //  const { to, subject, text } = req.body;
+//     const user= "st3196420@gmail.com"
+// debugger
+//     try {
+//         await transporter.sendMail({
+//             from: 'mazaltovchagit16@gmail.com',
+//             to: user,
+//             subject : "succsesssssss",
+//             text: "this is the kod"
+//         });
+//        console.log({ message: 'Email sent successfully!' });
+//     } catch (error) {
+//         console.error('Error sending email:', error);
+//         // res.status(500).json({ message: 'Failed to send email.' });
+//     }
+// };
+
+// module.exports = { sendEmail };
+// sendEmail({}, {});
+
+export const getUserByGoogleAccount =async (req, res) =>{
+  const user = getByEmail1 ( req.params.email)
+  if (user)
+    return res.status(200).send(user)
+  else{
+        const { tokenId } = req.params.token;
+        try {
+          const ticket = await client.verifyIdToken({
+            idToken: tokenId,
+            audience: '1079513751617-cvu5tvh0ggnogkvj8im50op7gocnehrj.apps.googleusercontent.com' // יש להחליף גם כאן במזהה הלקוח שלך
+          });
+          const payload = ticket.getPayload();
+          const { email, name, picture } = payload;
+          const token = jwt.sign({ email, name, picture }, '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', { expiresIn: '1h' }); // יש להחליף במפתח סודי עבור JWT שלך
+          res.json({
+            token,
+            userData: { email, name, picture }
+          });
+        } catch (error) {
+          console.error('Google login error:', error);
+          res.status(500).json({ error: 'Google login failed' });
+        }
+        {
+          
+        }
+      };
+
+  }
+
+  export const signIn = async (req, res, next) => {
+    try {
+      console.log(req.body,"klkl");
+      const { email, password } = req.body;
+      const user = await User.findOne({ email });
+      if (user) {
+        console.log('Password from request:', password);
+        console.log('Password from database:', user.password);
+        
+        bcrypt.compare( password, user.password, (err, same) => {
+          if (err) {
+            console.log("Error in bcrypt compare:", err);
+            return next(new Error(err.message));
```

Comment:
add status code

Reply:
it is not my function

Reply:
so why you have this changes in your branch?